esp32/sdkconfig.h

```diff
@@ -36,7 +44,6 @@
 #define CONFIG_TASK_WDT_CHECK_IDLE_TASK 0
 #define CONFIG_TASK_WDT_CHECK_IDLE_TASK_CPU1 0
 
-#define CONFIG_FREERTOS_UNICORE 1
```

Comment:
Dug into this:

Here's what you get if you restore `CONFIG_FREERTOS_UNICORE`:

```
/home/epoulsen/workspaces/esp-idf//components/esp32/dport_access.c: In function 'dport_access_init_core1':
/home/epoulsen/workspaces/esp-idf//components/esp32/dport_access.c:164:21: error: array subscript is above array bounds [-Werror=array-bounds]
     dport_access_ref[core_id] = 0;
                     ^
/home/epoulsen/workspaces/esp-idf//components/esp32/dport_access.c:165:23: error: array subscript is above array bounds [-Werror=array-bounds]
     dport_access_start[core_id] = 0;
                       ^
/home/epoulsen/workspaces/esp-idf//components/esp32/dport_access.c:166:21: error: array subscript is above array bounds [-Werror=array-bounds]
     dport_access_end[core_id] = 0;
                     ^
/home/epoulsen/workspaces/esp-idf//components/esp32/dport_access.c:167:21: error: array subscript is above array bounds [-Werror=array-bounds]
     dport_core_state[core_id] = DPORT_CORE_STATE_RUNNING;
                     ^
cc1: all warnings being treated as errors
make: *** [build//home/epoulsen/workspaces/esp-idf//components/esp32/dport_access.o] Error 1
make: *** Waiting for unfinished jobs....
epoulsen@banana ~/workspaces/micropython-esp32/esp32 $ 
```

Code in question:

Note that `dport_access_init_core1` is not guarded by any `#ifdef`.  Evidently the toolchain or the IDF code is set up to do static evaluation of `assert()` statements where possible.

esp-idf/components/esp32/dport_access.c:

```c
static void dport_access_init_core1(void *arg)
{
    int core_id = xPortGetCoreID();

    assert(core_id == 1);

    ESP_INTR_DISABLE(ETS_DPORT_INUM);
    intr_matrix_set(core_id, ETS_FROM_CPU_INTR3_SOURCE, ETS_DPORT_INUM);
    ESP_INTR_ENABLE(ETS_DPORT_INUM);

    dport_access_ref[core_id] = 0;
    dport_access_start[core_id] = 0;
    dport_access_end[core_id] = 0;
    dport_core_state[core_id] = DPORT_CORE_STATE_RUNNING;

    vTaskDelete(NULL);
}
```
esp-idf/components/freertos/include/freertos/FreeRTOSConfig.h: 

```c
/* ESP31 and ESP32 are dualcore processors. */
#ifndef CONFIG_FREERTOS_UNICORE
#define portNUM_PROCESSORS 2
#else 
#define portNUM_PROCESSORS 1
#endif
```

If you remove`esp-idf/components/esp32/dport_access.c` from the Makefile, you get a whole bunch of link errors:

```
LINK build/application.elf
build//home/epoulsen/workspaces/esp-idf//components/freertos/xtensa_vectors.o:(.iram1.literal+0x54): undefined reference to `dport_access_start'
build//home/epoulsen/workspaces/esp-idf//components/freertos/xtensa_vectors.o:(.iram1.literal+0x58): undefined reference to `dport_access_end'
build//home/epoulsen/workspaces/esp-idf//components/newlib/time.o:(.literal.esp_setup_time_syscalls+0xc): undefined reference to `rtc_time_get'
build//home/epoulsen/workspaces/esp-idf//components/newlib/time.o: In function `esp_setup_time_syscalls':

... over 100 more lines ...

```